modules/plot/include/opencv2/plot.hpp

```diff
@@ -0,0 +1,302 @@
+        Plot(cv::Mat _plotDataX, cv::Mat _plotDataY)
+        {
+            //f the matrix is not Nx1 or 1xN
+            if((_plotDataX.cols > 1 && _plotDataX.rows > 1) || (_plotDataY.cols > 1 && _plotDataY.rows > 1))
+            {
+                std::cout << "ERROR: Plot data must be a 1xN or Nx1 matrix." << std::endl;
+                exit(0);
+            }
+
+            //if the matrix type is not CV_64F
+            if(_plotDataX.type() != CV_64F || _plotDataY.type() != CV_64F)
+            {
+                std::cout << "ERROR: Plot data type must be double (CV_64F)." << std::endl;
+                exit(0);
+            }
+
+            //in case we have a row matrix than needs to be transposed
+            if(_plotDataX.cols > _plotDataX.rows)
+            {
+                _plotDataX = _plotDataX.t();
+            }
+            if(_plotDataY.cols > _plotDataY.rows)
+            {
+                _plotDataY = _plotDataY.t();
+            }
+
+            plotHelper(_plotDataX, _plotDataY);
+        }
+
+        //set functions
+        void setMinX(double _plotMinX)
+        {
+            plotMinX = _plotMinX;
+            plotMinX_plusZero = _plotMinX;
+        }
+        void setMaxX(double _plotMaxX)
+        {
+            plotMaxX = _plotMaxX;
+            plotMaxX_plusZero = _plotMaxX;
+        }
+        void setMinY(double _plotMinY)
+        {
+            plotMinY = _plotMinY;
+            plotMinY_plusZero = _plotMinY;
+        }
+        void setMaxY(double _plotMaxY)
+        {
+            plotMaxY = _plotMaxY;
+            plotMaxY_plusZero = _plotMaxY;
+        }
+        void setPlotLineWidth(int _plotLineWidth)
+        {
+            plotLineWidth=_plotLineWidth;
+        }
+        void setPlotLineColor(cv::Scalar _plotLineColor)
+        {
+            plotLineColor=_plotLineColor;
+        }
+        void setPlotBackgroundColor(cv::Scalar _plotBackgroundColor)
+        {
+            plotBackgroundColor=_plotBackgroundColor;
+        }
+        void setPlotAxisColor(cv::Scalar _plotAxisColor)
+        {
+            plotAxisColor=_plotAxisColor;
+        }
+        void setPlotGridColor(cv::Scalar _plotGridColor)
+        {
+            plotGridColor=_plotGridColor;
+        }
+        void setPlotTextColor(cv::Scalar _plotTextColor)
+        {
+            plotTextColor=_plotTextColor;
+        }
+        void setPlotSize(int _plotSizeWidth, int _plotSizeHeight)
+        {
+            if(_plotSizeWidth > 400)
+                plotSizeWidth = _plotSizeWidth;
+            else
+                plotSizeWidth = 400;
+
+            if(_plotSizeHeight > 300)
+                plotSizeHeight = _plotSizeHeight;
+            else
+                plotSizeHeight = 300;
+        }
+
+        //render the plotResult to a Mat
+        void render(cv::Mat &_plotResult);
+
+        //show the plotResult from within the class
+        void show(const char * _plotName);
```

Comment:
these 2 methods add unnecessary dependency of highgui and are not really needed. Please, remove them and let user explicitly call imread or imwrite.